cpp/libs/ossrf_gstreamer_api/lib/src/receiver/st2110_30_receiver_plugin.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2024 Advanced Media Workflow Association
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "st2110_30_receiver_plugin.h"
+#include "bisect/expected/macros.h"
+#include "bisect/pipeline.h"
+#include <gst/gst.h>
+
+using namespace bisect;
+using namespace ossrf::gst::receiver;
+using namespace ossrf::gst::plugins;
+
+namespace
+{
+    constexpr auto queue_max_size_time    = 200000;
+    constexpr auto queue_max_size_buffers = 0;
+    constexpr auto queue_max_size_bytes   = 0;
+}; // namespace
+
+struct gst_st2110_30_receiver_impl : gst_receiver_plugin_t
+{
+    receiver_settings s_;
+    audio_info_t f_;
+    gst::pipeline pipeline_;
+
+    gst_st2110_30_receiver_impl(receiver_settings settings, audio_info_t format) : s_(settings), f_(format) {}
+
+    ~gst_st2110_30_receiver_impl() { stop(); }
+
+    maybe_ok create_gstreamer_pipeline()
+    {
+        // Create pipeline and check if all elements are created successfully
+        BST_CHECK_ASSIGN(pipeline_, bisect::gst::pipeline::create(NULL));
+        auto* pipeline = pipeline_.get();
+
+        // Add pipeline udp source
+        auto* source = gst_element_factory_make("udpsrc", NULL);
+        BST_ENFORCE(source != nullptr, "Failed creating GStreamer element udpsrc");
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), source), "Failed adding udpsrc to the pipeline");
+
+        // Set udp source params
+        g_object_set(G_OBJECT(source), "address", s_.primary.source_ip_address.c_str(), NULL);
+        g_object_set(G_OBJECT(source), "auto-multicast", TRUE, NULL);
+        g_object_set(G_OBJECT(source), "port", s_.primary.source_port, NULL);
+        g_object_set(G_OBJECT(source), "multicast-iface", s_.primary.interface_name.c_str(), NULL);
+
+        // Create and set caps for udp source
+        GstCaps* caps = gst_caps_from_string("application/x-rtp, clock-rate=48000, channels=2");
```

Comment:
The `audio_info_t` has the information on the number of channels and clock rate to set on these caps, that is information parsed from the SDP